ark/utils/data_utils.py

```diff
@@ -46,7 +47,131 @@ def label_cells_by_cluster(fovs, all_data, label_maps, fov_col=settings.FOV_ID,
         labeled_img_array = label_maps.loc[label_maps.fovs == fov].squeeze().values
         relabeled_img_array = relabel_segmentation(labeled_img_array, labels_dict)
         img_data.append(relabeled_img_array)
-    np.stack(img_data, axis=0)
+
+    return xr.DataArray(img_data, coords=[fovs, range(img_data[0].shape[0]),
+                                          range(img_data[0].shape[1])],
+                        dims=["fovs", "rows", "cols"])
+
+
+def generate_cell_cluster_mask(fovs, base_dir, seg_dir, cell_consensus_name,
+                               cluster_col='cluster'):
+    """For each fov, create a mask labeling each cell with their SOM or meta cluster label
+
+    Args:
+        fovs (list):
+            List of fovs to relabel
+        base_dir (str):
+            The path to the data directory
+        seg_dir (str):
+            The path to the segmentation data
+        cell_consensus_name (str):
+            The path to the data with both cell SOM and meta cluster assignments
+        cluster_col (str):
+            Whether to assign SOM or meta clusters, needs to be 'cluster' or 'hCluster_cap'
+
+    Returns:
+        xarray.DataArray:
+            The labeled images (dims: ["fovs", "rows", "cols"])
+    """
+
+    # path checking
+    if not os.path.exists(seg_dir):
+        raise FileNotFoundError("seg_dir %s does not exist")
+
+    if not os.path.exists(os.path.join(base_dir, cell_consensus_name)):
+        raise FileNotFoundError(
+            "consensus_dir %s does not exist in base_dir %s" % (cell_consensus_name, base_dir)
+        )
+
+    # load the consensus data in
+    cell_consensus_data = feather.read_dataframe(os.path.join(base_dir, cell_consensus_name))
+
+    # define the files for whole cell and nuclear
+    whole_cell_files = [fov + '_feature_0.tif' for fov in fovs]
+    # nuclear_files = [fov + '_feature_1.tif' for fov in fovs]
+
+    # load the segmentation labels in
+    label_maps = load_utils.load_imgs_from_dir(data_dir=seg_dir,
+                                               files=whole_cell_files,
+                                               xr_dim_name='compartments',
+                                               xr_channel_names=['whole_cell'],
+                                               trim_suffix='_feature_0',
+                                               force_ints=True)
+
+    # use label_cells_by_cluster to create cell masks
```

Comment:
We can `verify` here that the supplied FOVs all appear in the `cell_consensus_data`